cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -1104,16 +1104,17 @@ writePoolState mOutFile serialisedCurrentEpochState = do
     pure (decodePoolState serialisedCurrentEpochState)
       & onLeft (left . QueryCmdPoolStateDecodeError)
 
-  let hks =
+  let hks :: [L.KeyHash L.StakePool StandardCrypto]
+      hks =
         toList $
-          Set.fromList $
+          fromList @(Set (L.KeyHash L.StakePool StandardCrypto)) $
```

Comment:
Is the type annotation needed? In this case, I'm not sure this is really an improvement, but fine nevertheless.

Reply:
Good catch. I thought we're going through set here to deduplicate elements - but the map keys are already unique, so it's pointless. I'll remove the conversion.